### ISSUE TITLE
Dock behavior unstable in 6.10(32)

### ISSUE THREAD

**Issue by victor0d** [TBog/TBLauncher#256]
Hi TBog, so many new things in this version. Including many improvements. THANK YOU. A few things here and there.

This bug is related to the Dock behaviour with different settings.

1) With **Features/Dock/Visible-for-results = 0** :
- **Behaviour/Desktop-mode-Widget/Show-Dock = 1** works
- **Behaviour/Desktop-mode-Search/Show-Dock = 1** works only if *Behaviour/Desktop-mode-Widget/Show-Dock = 1* (not if *Behaviour/Desktop-mode-Widget/Show-Dock = 0*).

2) With **Features/Dock/Visible-for-results = 1** :
- **Behaviour/Desktop-mode-Search/Show-Dock = 1** doesn't work
- **Behaviour/Desktop-mode-Widget/Show-Dock = 1** doesn't work
The DOCK appears and disappears with results (as expected) but Search_Desktop and Widgets_Desktop doesn't show the Dock.

If you need full settings copy and/or videos/screen-captures let me know.

**Comment by TBog** [comment]
When you set **Features | Dock | Visible for results** (visible when the result list has entries) you get to see the dock only when you have results and the result list is visible.
When you set **Behaviour | Desktop mode Search | Show Dock** you tell the Dock to show, but when result list is empty it can't be shown because of the **Visible for results** option.

I'm not sure what you expect from each option, but maybe you can help me rename them or add a description/summary for the preferences to better explain the function.

How would you implement **Visible for results** if not the way I did?

**Comment by victor0d** [comment]
Hi TBog,

The behaviour you are explaining is logical too. I misunderstood it. I apologize for that. I was expecting the Dock to show in any case when **Behaviour | Desktop mode Search | Show Dock** was set. SORRY.

Only as a suggestion to try to make it more clear:
. **Features | Dock | Visible only for results** or
.  **Features | Dock | Force visible only for results** (more clear but maybe too wordy :o)

The actual but is more the part-1. Unless I have misunderstood something else, it is a strange behaviour: 
. **Features | Dock | Visible for results** OFF
. **Behaviour | Desktop mode Search | Show Dock** ON
. **Behaviour | Desktop mode Widgets | Show Dock** OFF
The dock will not show in the Search desktop
(You may need to restart the launcher to reproduce it. Toggling **Behaviour | Desktop mode Widgets | Show Dock** ON and OFF, will make it work - this is the workaround :o).

**Comment by TBog** [comment]
Instead of **Visible only for results** maybe I should rename it as **Hidden when no results**

The issue you are having may be an issue I fixed today. I'll let you know when I make a new test build.

**Comment by TBog** [comment]
I still need to figure out what to do when both **Behaviour | Desktop mode Widgets | Show Dock** and **Features | Dock | Visible for results** are on.
Should the Dock be hidden because there are no results? If not then I need to move **Visible for results** to **Behaviour | Desktop mode Search**

**Comment by victor0d** [comment]
Hiding it would be consistent with your previous explanation.

Nevertheless, I think it is mostly a feature for the **Desktop Mode Search**. And moving it inside it (the **Desktop Mode Search**) will make sense. This one would be my preferred option. But this is only my point of view.

**Comment by TBog** [comment]
Can you please test this release: [v6.10-dev.1](https://github.com/TBog/TBLauncher/releases/tag/v6.10-dev.1)

**Comment by victor0d** [comment]
Works perfectly.
Totally resolved. And everything  very clear.
Thanks TBog.
This can be closed now.

**Comment by victor0d** [comment]
Let me check a bit longer. Maybe I am missing something.
Don't close it just now.

**Comment by victor0d** [comment]
Hi TBog, this is the most reliable way I have found (so far) to replicate it in this test version:
. **Behaviour | Desktop mode Search | Hidden when no results** OFF
. **Behaviour | Desktop mode Search | Show Dock** ON
. **Behaviour | Desktop mode Widgets | Show Dock** OFF

Works as expected.

But if now we toggle **Behaviour | Desktop mode Empty | Show Dock** OFF and ON (it may take two times toggling), the **Dock** will disappear in the **Search Desktop**.
It can be restored back (workaround) either (1) restarting the launcher (**Close Launcher**), or (2) toggling OFF and back ON **Features | Dock | Use Dock**).

I have experienced the issue under other circumstances but I had not been able to re-reproduce it in a systematic way. Twice, the issue appeared without modifying the settings.

I will add further information as I manage to re-reproduce it.

**Comment by TBog** [comment]
Do you have a completely transparent background for the **dock**? I'm asking because I think it may just be that the dock is empty (there was a bug like this at one point).

**Comment by victor0d** [comment]
Yes, that's correct TBog. Fully transparent dock.

If I remove the transparency of the Dock, the issue gets resolved.
And even if I set again the full transparency, I can't reproduce the issue anymore unless I restart the launcher.

**Comment by TBog** [comment]
I think I fixed the issue in commit 03500200222e09c0c8972a79372df48cc0ff329e

**Comment by victor0d** [comment]
Excellent TBog !!!

**Comment by TBog** [comment]
Is it fixed? Can we close this?

**Comment by victor0d** [comment]
Hi TBog. Sorry I had not been very active the last two weeks. I had been very busy.
This is becoming the best launcher in Android. No doubt.
Yes, it is sorted. Can be closed now !!!
I had tested it with and without dock transparency and works perfectly.

**Comment by victor0d** [comment]
Hi TBog,
I know we just have closed this issue. And it actually works perfectly!!! Nevertheless, I was reading the text toggle in the settings and I think we should make it cleared because it doesn't include the word DOCK:
`Hide when no results` (Visible when the result list has entries)
I think it maybe better adding the word DOCK, otherwise it seems unclear:
`Hide dock when no results` (Visible when the result list has entries)
